docs/docs/adding-a-shopping-cart-with-snipcart.md

```diff
@@ -0,0 +1,201 @@
+
+## Defining Products
+
+Adding products with Snipcart involves writing HTML representing your product and adding a set of attributes to that HTML. You might write something similar to the following code block for each item in your catalogue.
+
+```html
+<section>
+  <h2>Silver Stacking Ring</h2>
+  <p>$19.99</p>
+  <p>Wear one or seventeen! These rings are fun to mix and match.</p>
+</section>
+```
+
+Including this information allows a visitor to see what you have for sale, but they can't do anything with that information quite yet. You'll need a way for customers to add individual items to their cart. Try adding a button with the following attributes.
+
+```html
+<section>
+  <h2>Silver Stacking Ring</h2>
+  <p>$19.99</p>
+  <p>Wear one or seventeen! These rings are fun to mix and match.</p>
+  <!-- highlight-start -->
+  <button
+    class="snipcart-add-item"
+    data-item-id="silver-stacking-ring"
+    data-item-price="19.99"
+    data-item-url="/"
+    data-item-name="Silver Stacking Ring"
+  >
+    Add to cart
+  </button>
+  <!-- highlight-end -->
+</section>
+```
+
+Snipcart uses these attributes (`data-item-*`) to figure out what your customer is trying to buy and how much to charge them. The ID, price, URL, and name attributes are all required but there are several others that you can add to enhance the shopping cart.
+
+Importantly, `data-item-url` denotes the URL of the webpage displaying the product(s). Snipcart needs to crawl this page to validate the order. The web crawler looks for the HTML element with the `snipcart-add-item` CSS class as well as the `data-item-id` and checks what it finds there against whatever is in the cart.
+
+Note that, while you're testing, a `data-item-url` value of `"/"` is fine. For the checkout flow to work, you will eventually need to replace this with the actual URL at which you've published your catalogue or product page.
+
+To learn more about defining products, see the [Snipcart documentation](https://docs.snipcart.com/v3/setup/products).
+
+### Adding product variants
+
+Snipcart refers to variations like size and color as "product options" or "custom fields". You can add these custom fields to your products to allow customers to refine their orders.
+
+Building on the stacking ring example, suppose that you wanted to give your customer a choice between available sizes. You would do this by adding a custom field with a name and options.
+
+```html
+<section>
+  <h2>Silver Stacking Ring</h2>
+  <p>$19.99</p>
+  <p>Wear one or seventeen! These rings are fun to mix and match.</p>
+  <button
+    class="snipcart-add-item"
+    data-item-id="silver-stacking-ring"
+    data-item-price="19.99"
+    data-item-url="/"
+    data-item-name="Silver Stacking Ring"
+    <!--
+    highlight-start
+    --
+  >
+    data-item-custom1-name="Size"
+    data-item-custom1-options="6|6.5|7|7.5|8|8.5|9">
+    <!-- highlight-end -->
+    Add to cart
+  </button>
+</section>
```

Comment:
```suggestion
  <button
    class="snipcart-add-item"
    data-item-id="silver-stacking-ring"
    data-item-price="19.99"
    data-item-url="/"
    data-item-name="Silver Stacking Ring"
    <!--highlight-start-->
    data-item-custom1-name="Size"
    data-item-custom1-options="6|6.5|7|7.5|8|8.5|9"
    <!-- highlight-end -->
    >
    Add to cart
  </button>
</section>
```

Reply:
I'm not sure how that even happened! Prettier? Good catch!